### ISSUE TITLE
C# extension stop working after extension update

### ISSUE THREAD

**Issue by imtutta** [OmniSharp/omnisharp-vscode#3337]
## Issue Description ##
The C# extension stop working properly after an update.

## Steps to Reproduce ##
Just open VSCode

## Expected Behavior ##
C# project loaded correctly and intellisense working smoothly.

## Actual Behavior ##
No project loaded no intellisense

### OmniSharp log ###
```Starting OmniSharp server at 10/10/2019, 5:22:56 PM
    Target: /Users/matteo/stash/Golf_Rumble/stompy-client/stompy-client.sln

OmniSharp server started.
    Path: /Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/run
    PID: 16396

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.14.5 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 16.0 - "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/MSBuild.dll'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.0 - "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin"
            MSBuildExtensionsPath = /Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/xbuild
            BypassFrameworkInstallChecks = true
            CscToolPath = /Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/Roslyn
            CscToolExe = csc.exe
            MSBuildToolsPath = /Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin
            TargetFrameworkRootPath = /Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/xbuild-frameworks
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/matteo/stash/Golf_Rumble/stompy-client'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        Detecting projects in '/Users/matteo/stash/Golf_Rumble/stompy-client/stompy-client.sln'.
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp-Editor.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/matteo/stash/Golf_Rumble/stompy-client'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.EditorConfigWorkspaceOptionsProvider, Order: 200
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/matteo/stash/Golf_Rumble/stompy-client' on host 16345.
[fail]: OmniSharp.MSBuild.ProjectLoader
        The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj'.
/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj
/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets(1195,5): Error: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks

[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp-Editor.csproj
[warn]: OmniSharp.MSBuild.ProjectLoader
        "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/NuGet.targets" cannot be imported again. It was already imported at "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6060,3)". This is most likely a build authoring error. This subsequent import will be ignored. [/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj]
[fail]: OmniSharp.MSBuild.ProjectLoader
        The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp-Editor.csproj'.
/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp-Editor.csproj
/Library/Frameworks/Mono.framework/Versions/5.4.1/lib/mono/xbuild/15.0/Microsoft.Common.targets/ImportAfter/Microsoft.NuGet.ImportAfter.targets(21,3): Error: "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/NuGet.targets" cannot be imported again. It was already imported at "/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets (6060,3)". This is most likely a build authoring error. This subsequent import will be ignored. [/Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj]
/Users/matteo/.vscode/extensions/ms-vscode.csharp-1.21.4/.omnisharp/1.34.4/omnisharp/.msbuild/Current/Bin/Microsoft.Common.CurrentVersion.targets(1195,5): Error: The reference assemblies for .NETFramework,Version=v4.7.1 were not found. To resolve this, install the Developer Pack (SDK/Targeting Pack) for this framework version or retarget your application. You can download .NET Framework Developer Packs at https://aka.ms/msbuild/developerpacks

[fail]: OmniSharp.MSBuild.ProjectManager
        Attempted to update project that is not loaded: /Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp.csproj
[fail]: OmniSharp.MSBuild.ProjectManager
        Attempted to update project that is not loaded: /Users/matteo/stash/Golf_Rumble/stompy-client/Assembly-CSharp-Editor.csproj
```

## Environment information ##

**VSCode version**: 1.39.0
**C# Extension**: 1.21.4

<details><summary>Mono Information</summary>
        OmniSharp using built-in mono</details>
<details><summary>Dotnet Information</summary>
.NET Command Line Tools (1.1.11)

Product Information:
 Version:            1.1.11
 Commit SHA-1 hash:  d2610b5786

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.11-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.1.11
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-vscode|1.21.4|;

</details>

**Comment by CountessAnise** [comment]
I'm experiencing the same issue. Oddly enough, it's still working on my Windows machine.
Debian machine is a no go (same log as above).
Both machines running the same version (1.34.4, also same as above).

**Comment by winzig** [comment]
Same, had to go back to 1.21.3 to get it working again. I'm using vscode `1.40.0-insider`, fwiw.

**Comment by filipw** [comment]
according to your log you have Mono 5.4.1 on your machine, but the minimum is Mono 6.4.0 (to support .NET Core 3.0). Updating Mono should resolve the issue.

**Comment by imtutta** [comment]
Updating to Mono 6.4.0 fixed the issue. Thanks @filipw

**Comment by filipw** [comment]
thanks for confirming!
the latest version of the extension ships with msbuild 16.3 which is needed for .NET Core 3.0 support. however when you have Mono installed on your machine OmniSharp will try to use it. previously older Mono versions were fine, but now to work properly with OmniSharp Mono must have 16.3 msbuild in it, which is only available in Mono 6.4.0.

hope this explains the background a bit. we are working on improving the error messages around that to make things clearer.

**Comment by CountessAnise** [comment]
Updating Mono on my Debian machine did it for me as well.
Thank you @filipw
